ocaml/xapi/cli_operations.ml

```diff
@@ -2843,15 +2843,42 @@ let vm_migrate printer rpc session_id params =
              let gpu_group = Client.GPU_group.get_by_uuid remote_rpc remote_session gpu_group_uuid in
              vgpu,gpu_group) (read_map_params "vgpu" params) in
 
-         let default_sr =
-           try let pools = Client.Pool.get_all remote_rpc remote_session in
-             printer (Cli_printer.PMsg "Selecting remote pool's default SR for migrating VDIs") ;
-             Some (Client.Pool.get_default_SR remote_rpc remote_session (List.hd pools))
+         let preferred_sr =
+           (* The preferred SR is determined to be as the SR that the destine host has a PDB attached to it,
+              and among the choices of that the shared is preferred first(as it is recommended to have shared storage
+              in pool to host VMs), and then the one with the maximum available space *)
+           try
+             let query = Printf.sprintf {|(field "host"="%s") and (field "currently_attached"="true")|} (Ref.string_of host) in
+             let host_pbds = Client.PBD.get_all_records_where remote_rpc remote_session query in
+             let srs = List.map (fun (pbd_ref, pbd_rec) -> 
+               pbd_rec.API.pBD_SR, Client.SR.get_record remote_rpc remote_session pbd_rec.API.pBD_SR) host_pbds in
+             (* In the following loop, the current SR:sr' will be compared with previous checked ones, 
+                first if it is an ISO type, then pass this one for selection, then the only shared one from this and
+                prevoius one will be valued, and if not that case (both shared or none shared), choose the one with
+                more space available *)
+             let (sr, free_space) = List.fold_left (fun (sr, free_space) sr' ->
+               let sr_rec' = snd sr' in
```

Comment:
Instead of this, just do a more finegrained match in the function arguments, e.g.:

    List.fold_left (fun (sr, free_space) ((_, sr_rec') as sr') ->